src/js/jobs/components/__tests__/Step.test.js

```diff
@@ -1,29 +1,36 @@
 jest.mock("../../utils");
```

Comment:
As a general note for the tests I would suggest changing your tests to check the most common situations that can occur. To do this I would suggest modifying the passed props to mimic what would be passed when the component is being rendered in the browser. 

Of the top of my head this would be three main configurations. 
 1. The `step_name` and `step_description` is sent from the server. 
 2. No `step_name` or `step_description` but the `state` is one of the special cases ie: waiting
 3. No `step_name` or `step_description` and `state` is not one of the special cases.

Create props that mimic each of these situations then write separate tests for each situation that checks for correct rendering using RTL.